internal/pkg/endpoints/endpoints.go

```diff
@@ -99,14 +104,25 @@ func urlToTarget(URL string, TLSConfig TLSConfig) (Target, error) {
 		u.Path = "/metrics"
 	}
 
+	targetName := u.Host
+	if hostID != "" {
+		targetName = replaceLocalhost(u.Host, hostID)
+	}
+
 	return Target{
-		Name: u.Host,
+		Name: targetName,
 		Object: Object{
-			Name:   u.Host,
+			Name:   targetName,
 			Kind:   "user_provided",
 			Labels: make(labels.Set),
 		},
 		TLSConfig: TLSConfig,
 		URL:       *u,
 	}, nil
 }
+
+// ReplaceLocalhost replaces the occurrence of a localhost address with
+// the given hostname.
+func replaceLocalhost(source, with string) string {
+	return localhostReplaceRE.ReplaceAllString(source, with)
+}
```

Comment:
This is a copy of the function existing on the SDK. I Thought about asking CAOS to make it public so we can call it from here without redeclaring it. I see points in favor and against. First, right now this function leaves on the integration package of the SDK, optimally, if exporting it, I would ask CAOS to place it in a utils package or something. The point that I see against using an exported one is in the case that we needed the two functions to diverge. What do you thing @gsanchezgavier ?

Reply:
Oks, I added tests covering the different cases for replacement, we will stick to this local copy.